### PR TITLE
Search API: expose chunk matches over the streaming API

### PR DESCRIPTION
**Pull request by camdencheek**
This exposes chunk matches on the streaming API the URL query param `cm=t`. 

It is currently unused, but I added a test to ensure chunk matches actually come through. Will experiment a bit to see how much work it will be to update the client side of things. 

Note that Zoekt still doesn't send chunk matches, and I probably won't flip the switch here until it does.

## Test plan

Added a unit test. Will be exercised more when the web client is updated to use it.

### CODE REVIEW

File: cmd/frontend/internal/search/search.go [hunk 1]
```diff
@@ -234,6 +236,11 @@ func parseURLQuery(q url.Values) (*args, error) {
 		return nil, errors.Errorf("display must be an integer, got %q: %w", display, err)
 	}
 
+	chunkMatches := get("cm", "f")
+	if a.ChunkMatches, err = strconv.ParseBool(chunkMatches); err != nil {
+		return nil, errors.Errorf("chunk matches must be parseable as a boolean, got %q: w", chunkMatches, err)
```
Comment (tbliu98) [inline cmd/frontend/internal/search/search.go:241]:
```suggestion
		return nil, errors.Errorf("chunk matches must be parseable as a boolean, got %q: %w", chunkMatches, err)
```

**Review by tbliu98 — APPROVED** [review]
Just a few small things. Also apologies, I somehow completely missed this coming through my inbox

File: cmd/frontend/internal/search/search.go (hunk 1 above)
Comment (camdencheek) [reply cmd/frontend/internal/search/search.go:241]: whoops, thanks!

File: cmd/frontend/internal/search/search.go [hunk 2]
```diff
@@ -195,10 +196,11 @@ func logSearch(ctx context.Context, alert *search.Alert, err error, start time.T
...
 type args struct {
-	Query       string
-	Version     string
-	PatternType string
-	Display     int
+	Query        string
+	Version      string
+	PatternType  string
+	Display      int
+	ChunkMatches bool
```
[the remark this answers is not in the document]
Comment (camdencheek) [reply cmd/frontend/internal/search/search.go:203]: oh haha, I forgot I did that

**Comment by camdencheek** [comment]
FYI @bobheadxi  (for Raycast) and @vdavid (for IntelliJ integration), chunk matches will be the direction we're moving on the streaming API and we will eventually deprecate and remove line matches.

**Pull request merged**